src/connection-manager/index.ts

```diff
@@ -167,8 +173,18 @@ export class DefaultConnectionManager extends EventEmitter<ConnectionManagerEven
 
     this.opts = mergeOptions.call({ ignoreUndefined: true }, defaultOptions, init)
 
-    if (this.opts.maxConnections < this.opts.minConnections) {
-      throw errCode(new Error('Connection Manager maxConnections must be greater than minConnections'), codes.ERR_INVALID_PARAMETERS)
+    if (this.opts.maxIncomingConnections < this.opts.minConnections) {
+      throw errCode(
+        new Error('Connection Manager maxIncomingConnections must be greater than minConnections'),
+        codes.ERR_INVALID_PARAMETERS
+      )
+    }
+
+    if (this.opts.maxOutgoingConnections < this.opts.minConnections) {
```

Comment:
Shouldn't this condition be `this.opts.maxIncomingConnections + this.opts.maxOutgoingConnections < this.opts.minConnection` ?

Reply:
Good point, I was originally thinking that we would at the very least want both to be above the minimum connections but I suppose we would want to have more outgoing connections in some cases